src/modules/function/invocation_utils.rs

```diff
@@ -40,7 +40,7 @@ fn run_function_with_args(meta: &mut ParserMetadata, mut fun: FunctionDecl, args
     // Check if the function argument types match
     if fun.is_args_typed {
         for (index, (arg_name, arg_type, given_type)) in izip!(fun.arg_names.iter(), fun.arg_types.iter(), args.iter()).enumerate() {
-            if arg_type != given_type {
+            if given_type != arg_type && !given_type.is_subset_of(arg_type) {
```

Comment:
I still think that we should create a separate function that checks if this type can be of the other type (is equal or is a subset), basically what I wrote in the previous CR. It's so that we don't repeat ourselves ex. when a new such if statement is required, then we don't forget to check either condition (equality or subset).